components/bsx/Create/CreateCollection.vue

```diff
@@ -29,176 +29,136 @@
+const BaseCollectionForm = () =>
+  import('@/components/base/BaseCollectionForm.vue')
+const SubmitButton = () => import('@/components/base/SubmitButton.vue')
+const Money = () => import('@/components/bsx/format/TokenMoney.vue')
+const AccountBalance = () => import('@/components/shared/AccountBalance.vue')
+const MultiPaymentFeeButton = () =>
+  import('@/components/bsx/specific/MultiPaymentFeeButton.vue')
 
-  get balanceNotEnoughMessage() {
-    if (this.balanceNotEnough) {
-      return this.$t('tooltip.notEnoughBalance')
-    }
-    return ''
-  }
+const base = ref<BaseCollectionType>({
+  name: '',
+  file: null,
+  description: '',
+})
+const collectionDeposit = ref('')
+const attributes = ref<Attribute[]>([])
+const balanceNotEnough = ref(false)
+const feesToken = ref<Token>('BSX')
+const collectionForm = ref()
 
-  @Watch('accountId', { immediate: true })
-  async onAccountIdChange(val: string, oldVal: string) {
-    if (shouldUpdate(val, oldVal)) {
-      this.feesToken = await getFeesToken()
-    }
-  }
+onApiConnect(apiUrl.value, (api) => {
+  const classDeposit = getclassDeposit(api)
+  const metadataDeposit = getMetadataDeposit(api)
+  collectionDeposit.value = (classDeposit + metadataDeposit).toString()
+})
 
-  public async created() {
-    onApiConnect(this.apiUrl, (api) => {
-      const classDeposit = getclassDeposit(api)
-      const metadataDeposit = getMetadataDeposit(api)
-      this.collectionDeposit = (classDeposit + metadataDeposit).toString()
-    })
-  }
-  get balanceOfToken() {
-    return getBalance(this.feesToken)
-  }
+function checkValidity() {
+  return collectionForm.value?.checkValidity()
+}
 
-  get depositOfToken() {
-    return getDeposit(this.feesToken, parseFloat(this.collectionDeposit))
+const balanceNotEnoughMessage = computed(() => {
+  if (balanceNotEnough.value) {
+    return $i18n.t('tooltip.notEnoughBalance')
   }
+  return ''
+})
 
-  protected async tryToEstimateTx(): Promise<string> {
-    const api = await ApiFactory.useApiInstance(this.apiUrl)
-    const cb = api.tx.utility.batchAll
-    const metadata = dummyIpfsCid()
-    const randomId = 0
-    const args = [createArgs(randomId, metadata)]
-    return estimate(this.accountId, cb, args)
-  }
+const balanceOfToken = computed(() => getBalance(feesToken.value))
+const depositOfToken = computed(() =>
+  getDeposit(feesToken.value, parseFloat(collectionDeposit.value))
+)
 
-  protected async checkBalanceBeforeTx(): Promise<void> {
-    const estimated = await this.tryToEstimateTx()
-    const deposit = this.collectionDeposit
-    const hasTokens = hasEnoughToken(this.balance, estimated, deposit)
-    this.$consola.log('hasTokens', hasTokens)
-    if (!hasTokens) {
-      throw new Error(
-        `Not enough tokens: Currently have ${formatBalance(
-          this.balance,
-          this.decimals,
-          this.unit
-        )} tokens`
-      )
-    }
+async function submit(): Promise<void> {
```

Comment:
i wouldn't advice having watch inside this function
It would register it everytime the user clicks submit

Reply:
What's the problem with "registering it every time the user clicks submit"? As I understand it, that's the main purpose of this process - to get a new transaction status after each user submission, rather than having to manually clean it up.